File: tests/fixtures/ocaml/sample.ml
open Printf
open List

let pi = 3.14159265358979

let greet name =
  printf "Hello, %s!\n" name

let add a b = a + b

let square x = x * x

let apply = fun f x -> f x

type shape =
  | Circle of float
  | Rectangle of float * float

type color = Red | Green | Blue

let area s =
  match s with
  | Circle r -> pi *. r *. r
  | Rectangle (w, h) -> w *. h

module MathUtils = struct
  let clamp value min_val max_val =
    if value < min_val then min_val
    else if value > max_val then max_val
    else value
end

module type Printable = sig
  val to_string : 'a -> string
end

let main () =
  let result = add 1 2 in
  printf "%d\n" result

File: tests/fixtures/zig/sample.zig
const std = @import("std");
const math = @import("std").math;

pub fn greet(name: []const u8) void {
    std.debug.print("Hello, {s}!\n", .{name});
}

pub fn add(a: i32, b: i32) i32 {
    return a + b;
}

pub fn square(x: i32) i32 {
    return x * x;
}

pub const Point = struct {
    x: f64,
    y: f64,

    pub fn distance(self: Point, other: Point) f64 {
        const dx = self.x - other.x;
        const dy = self.y - other.y;
        return math.sqrt(dx * dx + dy * dy);
    }
};

pub const Color = enum {
    Red,
    Green,
    Blue,
};

test "add returns correct sum" {
    try std.testing.expectEqual(@as(i32, 5), add(2, 3));
}
